components/movementsensor/rtkutils/cachedData.go

```diff
@@ -174,3 +180,49 @@ func (g *CachedData) Properties(
 		CompassHeadingSupported: true,
 	}, nil
 }
+
+// findDistance calculates the distance between two points on Earth.
+// lat1, lon1: Latitude and Longitude of point 1 (in decimal degrees)
+// lat2, lon2: Latitude and Longitude of point 2 (in decimal degrees)
+// returns the distance in meters.
+func findDistance(lat1, lon1, lat2, lon2 float64) float64 {
+	lat1Rad, lon1Rad := utils.DegToRad(lat1), utils.DegToRad(lon1)
+	lat2Rad, lon2Rad := utils.DegToRad(lat2), utils.DegToRad(lon2)
+
+	dLat := lat2Rad - lat1Rad
+	dLon := lon2Rad - lon1Rad
+
+	a := math.Sin(dLat/2)*math.Sin(dLat/2) +
+		math.Cos(lat1Rad)*math.Cos(lat2Rad)*
+			math.Sin(dLon/2)*math.Sin(dLon/2)
+	c := 2 * math.Atan2(math.Sqrt(a), math.Sqrt(1-a))
+
+	distanceKm := earthRadiusKm * c
+	return distanceKm * 1000 // convert km to meters
+}
+
+// calculateCompassDegreeError calculates the compass degree error
+// of two geo points.
+func (g *CachedData) calculateCompassDegreeError() float64 {
+	firstPos := g.lastPosition.GetLastPosition()
```

Comment:
This can only be non-nil after someone calls `Position()` on this sensor. If you don't open app.viam.com (e.g., because you're interacting with the robot solely through a Python SDK or something), that function will never be called, so this will always be nil, and we'll always return NaN. That strikes me as the wrong behavior, but I don't know what the right behavior would be.

Reply:
pass two geopoints in. Use the helper function from the drivers.